README.md

```diff
@@ -164,61 +170,69 @@ You can verify the installation by running:
 
 ## How to develop
 
-This assumes that you have rust and cargo installed. We use the workflow recommended by [pyo3](https://github.com/PyO3/pyo3) and [maturin](https://github.com/PyO3/maturin).
+This assumes that you have rust and cargo installed. We use the workflow recommended by [pyo3](https://github.com/PyO3/pyo3) and [maturin](https://github.com/PyO3/maturin). The Maturin tools used in this workflow can be installed either via `uv` or `pip`. Both approaches should offer the same experience. It is recommended to use `uv` since it has significant performance improvements
+over `pip`.
 
-The Maturin tools used in this workflow can be installed either via Conda or Pip. Both approaches should offer the same experience. Multiple approaches are only offered to appease developer preference. Bootstrapping for both Conda and Pip are as follows.
+Bootstrap (`uv`):
 
-Bootstrap (Conda):
+By default `uv` will attempt to build the datafusion python package. For our development we prefer to build manually. This means
+that when creating your virtual environment using `uv sync` you need to pass in the additional `--no-install-package datafusion`
+and for `uv run` commands the additional parameter `--no-project`
 
 ```bash
 # fetch this repo
 git clone git@github.com:apache/datafusion-python.git
-# create the conda environment for dev
-conda env create -f ./conda/environments/datafusion-dev.yaml -n datafusion-dev
-# activate the conda environment
-conda activate datafusion-dev
+# create the virtual enviornment
+uv sync --dev --no-install-package datafusion
+# activate the environment
+source .venv/bin/activate
 ```
 
-Or alternatively, if you are on an OS that supports CUDA Toolkit, you can use `-f ./conda/environments/datafusion-cuda-dev.yaml`.
-
-Bootstrap (Pip):
+Bootstrap (`pip`):
 
 ```bash
 # fetch this repo
 git clone git@github.com:apache/datafusion-python.git
 # prepare development environment (used to build wheel / install in development)
-python3 -m venv venv
+python3 -m venv .venv
 # activate the venv
-source venv/bin/activate
+source .venv/bin/activate
 # update pip itself if necessary
 python -m pip install -U pip
-# install dependencies (for Python 3.8+)
-python -m pip install -r requirements.in
+# install dependencies
+python -m pip install -r pyproject.toml
```

Comment:
You can just `pip install .` too. That will read the pep 517-compliant build backend and install dependencies and build datafusion